op-node/rollup/derive/batches.go

```diff
@@ -99,6 +99,11 @@ func checkSingularBatch(cfg *rollup.Config, log log.Logger, l1Blocks []eth.L1Blo
 		return BatchDrop
 	}
 
+	if cfg.IsInteropActivationBlock(batch.Timestamp) && len(batch.Transactions) >= 0 {
+		log.Warn("Interop activation block may not contain non-deposit transactions")
+		return BatchDrop
+	}
```

Comment:
Slight preference to move this check further down to group it with other checks of `batch.Transactions`, e.g. to line 168. So have the batch metadata checks like timestamp etc before checking content like transactions, going "from outside to inside".